rules/build_test.bzl

```
@@ -16,16 +16,44 @@
 
 load("//lib:new_sets.bzl", "sets")
 
+def _empty_test_impl(ctx):
+    is_windows = select({
+        "@bazel_tools//src/conditions:host_windows": True,
+        "//conditions:default": False,
+    })
+    extension = ".bat" if is_windows else ".sh"
+    executable = ctx.actions.declare_file(ctx.label.name + extension)
+    ctx.actions.write(
+        output = executable,
+        is_executable = True,
+        content = "exit 0",
+    )
+
+    runfiles = []
+    for entry in ctx.attr.data:
+        runfiles += entry.files.to_list()
+
+    return [DefaultInfo(
+        files = depset([executable]),
+        executable = executable,
+        runfiles = ctx.runfiles(files = runfiles),
+    )]
+
+_empty_test = rule(
+    implementation = _empty_test_impl,
+    attrs = {
+        "data": attr.label_list(),
```

```suggestion
        "data": attr.label_list(allow_files = True),
```
You can use the `allow_files` to make the code above simpler. No need to [call to_list() in a loop](https://docs.bazel.build/versions/master/skylark/depsets.html#performance)